domain/ide/workspace/folder_test.go

```diff
@@ -463,25 +465,51 @@ func Test_processResults_ShouldSendAnalyticsToAPI(t *testing.T) {
 		Issues:  []snyk.Issue{mockCodeIssue},
 	}
 
-	engineMock.EXPECT().GetConfiguration().AnyTimes().Return(gafConfig)
-	engineMock.EXPECT().InvokeWithInputAndConfig(localworkflows.WORKFLOWID_REPORT_ANALYTICS, gomock.Any(), gomock.Any()).
-		// this captures the call parameters of the mocked call
-		Do(func(id workflow.Identifier, workflowInputData []workflow.Data, config configuration.Configuration) {
-			require.Equal(t, 1, len(workflowInputData))
-			payloadBytes, ok := workflowInputData[0].GetPayload().([]byte)
-			require.True(t, ok)
-
-			var scanDoneEvent json_schemas.ScanDoneEvent
-			err := json.Unmarshal(payloadBytes, &scanDoneEvent)
+	ic := analytics.NewInstrumentationCollector()
+
+	ua := util.GetUserAgent(gafConfig, config.Version)
+	ic.SetUserAgent(ua)
+	categories := setupCategories(&data, c)
+	ic.SetCategory(categories)
+	ic.SetStage("dev")
+	ic.SetStatus("Success") //or get result status from scan
+	ic.SetInteractionType("Scan done")
+	summary := createTestSummary(&data, c)
+	ic.SetTestSummary(summary)
+	ic.SetType("Analytics")
+
+	entered := make(chan struct{})
```

Comment:
Alternative option: chan bool. That's what I usually use - don't think there's a big difference, though.

Reply:
I think the struct{} should be 0-sized afaik.